pink_balancer/wheel_controller.py

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 Stéphane Caron
+# Copyright 2023 Inria
+
+"""Control wheels to track ground and yaw velocity targets."""
+
+import gin
+import numpy as np
+from upkie.utils.clamp import clamp_abs
+from upkie.utils.filters import abs_bounded_derivative_filter
+
+from .sagittal_balance import MPCBalancer, PIBalancer, SagittalBalancer
+
+
+@gin.configurable
+class WheelController:
+    """
+    Base class for wheel balancers.
+
+    Attributes:
+        ground_velocity: Sagittal velocity in [m] / [s].
+        integral_error_velocity: Integral term contributing to the sagittal
+            velocity, in [m] / [s].
+        max_target_accel: Maximum acceleration for the ground target, in
+            [m] / [s]². Does not affect the commanded ground velocity.
+        max_target_velocity: Maximum velocity for the ground target, in
+            [m] / [s]. Indirectly affects the commanded ground velocity.
+        target_ground_velocity: Target ground sagittal velocity in [m] / [s].
+        target_yaw_velocity: Target yaw velocity in [rad] / [s].
+        turning_deadband: Joystick axis value between 0.0 and 1.0 below which
+            legs stiffen but the turning motion doesn't start.
+        turning_probability: Probability that the user wants to turn based on
+            the joystick axis value.
+        turning_decision_time: Minimum duration in [s] for the turning
+            probability to switch from zero to one and converesly.
+        wheel_radius: Wheel radius in [m].
+    """
+
+    ground_velocity: float
+    integral_error_velocity: float
+    max_target_accel: float
+    max_target_velocity: float
+    sagittal_balancer: SagittalBalancer
+    target_ground_velocity: float
+    target_yaw_velocity: float
+    turning_deadband: float
+    turning_decision_time: float
+    turning_probability: float
+    wheel_radius: float
+
+    def __init__(
+        self,
+        balancer_class: str,
```

Comment:
```suggestion
        balancer_class: Literal["MPCBalancer", "PIBalancer"],
```
Do you think it makes sense to use Literal types here ?

Reply:
Sure, why not. I didn't know about these types, thanks :smiley: